src/vivarium/framework/population/population_view.py

```diff
@@ -126,29 +131,33 @@ def update(self, population_update: pd.Series[Any] | pd.DataFrame) -> None:
             this view manages or if the view is being updated with a data
             type inconsistent with the original population data.
         """
-        state_table = self._manager.population.copy()
-        population_update = self._format_update_and_check_preconditions(
+        if not isinstance(self._manager.population, pd.DataFrame):
+            raise PopulationError(
+                "Trying to update a population that has not yet been initialized."
+            )
+        private_data = self._manager.get_private_data(self._component)
+        population_update_df: pd.DataFrame = self._format_update_and_check_preconditions(
             population_update,
-            state_table,
+            private_data,
             self.private_columns,
             self._manager.creating_initial_population,
             self._manager.adding_simulants,
         )
         if self._manager.creating_initial_population:
-            new_columns = list(set(population_update).difference(state_table))
-            self._manager.population[new_columns] = population_update[new_columns]
-        elif not population_update.empty:
-            update_columns = list(set(population_update).intersection(state_table))
+            new_columns = list(set(population_update_df).difference(private_data))
+            self._manager.population[new_columns] = population_update_df[new_columns]
+        elif not population_update_df.empty:
+            update_columns = list(set(population_update_df).intersection(private_data))
             for column in update_columns:
                 column_update = self._update_column_and_ensure_dtype(
-                    population_update[column],
-                    state_table[column],
+                    population_update_df[column],
+                    private_data[column],
                     self._manager.adding_simulants,
                 )
                 self._manager.population[column] = column_update
```

Comment:
I think it's a bit better to have a method `manager.update()` rather than have the view do the update itself.

Also, you can loop through all the columns and concat into a DF for a single update.